File: src/main/scala/com/zooplus/jacekb/learningTime/akka/Listener.scala
package com.zooplus.jacekb.learningTime.akka

import akka.actor.Actor
import com.zooplus.jacekb.learningTime.akka.common.Commons
import Commons.PiApproximation

/**
 * Created with IntelliJ IDEA.
 * User: jacek_bilski
 * Date: 28/06/13
 * Time: 09:23
 */
class Listener extends Actor {
	def receive = {
		case PiApproximation(pi, duration) ⇒
			println(s"\n\tPi approximation: \t$pi\n\tCalculation time: \t$duration")
			context.system.shutdown()
	}
}

File: src/main/java/com/zooplus/jacekb/learningTime/akka/pi/java/Listener.java
package com.zooplus.jacekb.learningTime.akka.pi.java;

import akka.actor.AbstractActor;
import akka.japi.pf.ReceiveBuilder;

/**
 * Created with IntelliJ IDEA.
 * User: jacek_bilski
 * Date: 18.09.14
 * Time: 14:56
 */
public class Listener extends AbstractActor {
    public Listener() {
        receive(ReceiveBuilder.match(PiApproximation.class, this::piApproximation).build());
    }

    private void piApproximation(PiApproximation piApproximation) {
        System.out.println("\n\tPi approximation: \t" + piApproximation.getPi() + "\n\tCalculation time: \t" + piApproximation.getDuration() + " ms");
        getContext().system().shutdown();
    }
}
